### DISCUSSION TITLE
@next/font in pages directory

### DISCUSSION THREAD

**Discussion by Code-Victor** [vercel/next.js#41849] [Help]
I hope to use the new `@next/font` package but still use the `pages` file structure. I tried adding the `fontFamily.className`  to the `Html` component in the `_document.tsx` file but I'm getting an error.
![image](https://user-images.githubusercontent.com/63613078/198015802-586cebd6-a56a-4536-9647-38dc625798ee.png)
is there any workaround?

**Comment by Code-Victor** [comment]
I have got a workaround. I added it in the `Head` tag of a global layout component.
```tsx
const inter = Inter();
// If loading a variable font, you don't need to specify the font weight
function Layout({ children }: { children: ReactNode }) {
  return (
    <>
      <Head>
        <style
          id="font"
          dangerouslySetInnerHTML={{
            __html: `*{font-family: ${inter.style.fontFamily}}`,
          }}
        />
      </Head>
      {children}
    </>
  );
}
```

**Reply by hanneslund** [reply]
If you want to use a font globally In `pages/` you can either add it on a wrapper in `_app`.
```tsx
// _app.tsx
import { Inter } from '@next/font/google'

const inter = Inter({ subsets: ['latin'] })

export default function MyApp({ Component, pageProps }) {
  return (
    <main className={inter.className}>
      <Component {...pageProps} />
    </main>
  )
}
```

Or if you don't want a wrapper, inject it to `<head>`.
```tsx
// _app.tsx
import { Inter } from '@next/font/google'

const inter = Inter({ subsets: ['latin'] })

export default function MyApp({ Component, pageProps }) {
  return (
    <>
      <style jsx global>{`
        html {
          font-family: ${inter.style.fontFamily};
        }
      `}</style>
      <Component {...pageProps} />
    </>
  )
}

```

More examples are coming to the docs.

**Reply by Code-Victor** [reply]
Thanks for this👍

**Reply by expert-47** [reply]
hi @hanneslund 

I have tried both ways but still unable to integrate inter font , 

can you please guide what maybe the issue ? 
here is my code in _app.tsx, I am providing the complete code , maybe some other file is causing the issue ?  
import { Inter } from "@next/font/google";
const inter = Inter({ subsets: ["latin"] });


    <CacheProvider value={emotionCache}>
      <Script
        id="gtag-base"
        strategy="lazyOnload"
        dangerouslySetInnerHTML={{
          __html: `
        var loadGtm = function(w, d, s, l, i) {w[l] = w[l] || [];w[l].push({'gtm.start': new Date().getTime(),event: 'gtm.js'});var f = d.getElementsByTagName(s)[0],
      j = d.createElement(s),
      dl = l != 'dataLayer' ? '&l=' + l : '';
  j.async = true;
  j.src = 'https://www.googletagmanager.com/gtm.js?id=' + i + dl;
  f.parentNode.insertBefore(j, f);
}
setTimeout(loadGtm.bind(null, window, document, 'script', 'dataLayer', '${process.env.NEXT_PUBLIC_GTM_KEY}'), 4000);'${process.env.NEXT_PUBLIC_GTM_KEY}'`,
        }}
      />
      <DefaultSeo {...SEO} />
      <Provider store={store}>
        <ApolloProvider client={client}>
          <ThemeProvider theme={theme}>
            <PersistGate persistor={persistor}>
              <NextNProgress color="#29D" startPosition={0.3} height={3} showOnShallow={true} />

              <main className={inter.className}>
                <Component {...pageProps} />
              </main>
            </PersistGate>
          </ThemeProvider>
        </ApolloProvider>
      </Provider>
    </CacheProvider>

**Comment by AhmedBHameed** [comment]
I have the same issue.
Seems we need more clarification about how it is working.

The part of [new feature -> @next/font](https://nextjs.org/blog/next-13#nextfont) needs more love maybe?

It is also worth to mention, in case of importing more than one weight of the same font, how to use it then? Should be createing a new variable for each weight?

From the type:
```ts
export declare function Roboto(options: {
    weight: '100' | '300' | '400' | '500' | '700' | '900';  <<== how with multi weights?
    style?: 'normal' | 'italic';
    display?: Display;
    variable?: CssVariable;
    preload?: boolean;
    fallback?: string[];
    adjustFontFallback?: boolean;
    subsets?: Array<'cyrillic' | 'cyrillic-ext' | 'greek' | 'greek-ext' | 'latin' | 'latin-ext' | 'vietnamese'>;
}): FontModule;
```

**Reply by haykam821** [reply]
It seems like the documentation wasn't published, since the compiler warnings have links that point to missing documentation:

> `warn` - The @next/font/google font Ubuntu has no selected subsets. Please specify subsets in the function call or in your next.config.js, otherwise no fonts will be preloaded. Read more: https://nextjs.org/docs/api-reference/components/font#nextfontgoogle

**Reply by hanneslund** [reply]
@haykam821 
Thanks! Docs are being worked on. You can find the beta docs [here](https://beta.nextjs.org/docs/optimizing/fonts).

@AhmedBHameed 
We're working on making it easier adding multiple weights. In the meantime, preferably you can use the variable font `Roboto Flex`, or using a different variable for each weight.

**Reply by hanneslund** [reply]
Multiple weights and styles are now available in `13.0.1` https://github.com/vercel/next.js/issues/41969#issuecomment-1298314889
